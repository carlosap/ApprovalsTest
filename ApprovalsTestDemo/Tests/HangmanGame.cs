using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using HangmanGame;
using NUnit.Framework;
using NUnit.Framework.Internal;

namespace Tests
{
    [TestFixture]
    public class HangmanGame
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GameState()
        {
            var sb = new StringBuilder();

            var game = new GameEngine("Microsoft");
            game.Guess('i');
            game.Guess('y');
            game.Guess('t');

            //1st check the state before the first letter s
            sb.AppendLine(game.ToString());

            game.Guess('s');

            //2nd check for the state after the letter s
            sb.AppendLine(game.ToString());

            Approvals.Verify(sb.ToString());
        }
    }
}
