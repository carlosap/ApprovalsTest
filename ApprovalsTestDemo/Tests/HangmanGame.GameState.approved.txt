Guessed letters: IYT
Guesses remaining: 9
MICROSOFT
_I______T

Guessed letters: IYTS
Guesses remaining: 9
MICROSOFT
_I___S__T

